core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2829,6 +2829,20 @@ public static long bitAnd(long b0, long b1) {
     return b0 & b1;
   }
 
+  /** Bitwise function <code>BITAND</code> applied to a Long and int value.
+   *  Needed for handling NULL for the first argument.
+   */
+  public static long bitAnd(Long b0, int b1) {
+    return b0 & b1;
+  }
+
+  /** Bitwise function <code>BITAND</code> applied to a Long and int value.
```

Comment:
Bitwise function <code>BITAND</code> applied to a int and Long value.

Reply:
Added some more tests with different integer types for the arguments (integer and bigint).

Needed to rework the return type detection. It will now use the largest integer type if both arguments are integer types. There may be ways of cleaning the code up further for the return type inference.